### PR TITLE
Fix compatibility issue with 'cp' command in shell script.

### PR DESCRIPTION
**Pull request by sebastiancarlos**
## Summary:

This change addresses a compatibility issue arising from the use of the macOS-specific `-X` flag in the `cp` command withing a shell script called during the `npm run ios` command.

The `-X` flag prevents the copying of Extended Attributes and resource forks, which indeed prevents possible issues on macOS. However the flag is not supported by GNU's `cp`, commonly installed via homebrew on macOS systems, and usually added to the user's PATH.

This fix basically does a feature test to figure out if the `-X` flag is available, and only uses it is the test is positive.

The intended behavior of not copying Extended Attributes and resources forks is still preserved in the script even if GNU's `cp` is in the path, as GNU's `cp` doesn't copy these by default.

In short, the script will behave correctly regardless of the `cp` version installed, but the script will now be more POSIX compatible.

The fix helps prevent unnecessary debugging time as, in my experience, the source of this error is not always displayed in the output of `npm run ios`.

As mentioned in: 
- https://github.com/facebook/react-native/issues/42112#issuecomment-1922990774
- https://github.com/facebook/react-native/issues/39734#issuecomment-1932644315

### Relevant Documentation
- [GNU's cp documentation](https://www.gnu.org/software/coreutils/manual/html_node/cp-invocation.html#cp-invocation)
- [macOS's cp man page](https://ss64.com/mac/cp.html)

## Changelog:

[IOS] [FIXED] - Fix compatibility issue with 'cp' command in shell script.

## Test Plan:

```
# with GNU's cp in path
npm run ios
# observe build passing (this is the fix)

# remove GNU's cp from path
npm run ios
# observe build passing (no regression)
```

### CODE REVIEW

**Comment by felixmueller-cnnctvstn** [comment]
Is there a tmp fix for that?

File: packages/react-native/scripts/react_native_pods_utils/script_phases.sh [hunk 1]
```diff
@@ -103,8 +103,17 @@ generateArtifacts () {
 moveOutputs () {
     mkdir -p "$RCT_SCRIPT_OUTPUT_DIR"
 
+    # Test if 'cp' supports the '-X' flag (GNU's cp doesn't, but macOS's cp does)
+    touch /tmp/test-cp-X-flag-one /tmp/test-cp-X-flag-two
+    if cp -X /tmp/test-cp-X-flag-one /tmp/test-cp-X-flag-two > /dev/null 2>&1; then
```
Comment (huntie) [inline packages/react-native/scripts/react_native_pods_utils/script_phases.sh:108]: Thanks for this change! 🙂

Can we test by `[[ "$OSTYPE" == "darwin"* ]]` instead? I appreciate this does direct feature detection, but would prefer if we avoided the file system side effects.

Comment (sebastiancarlos) [reply packages/react-native/scripts/react_native_pods_utils/script_phases.sh:108]: Hi @huntie. I agree that a way of feature detection without file systems side effect is desireable.

> Can we test by [[ "$OSTYPE" == "darwin"* ]] instead?

Unfortunately the `OSTYPE` detection is not enough, as it identifies the OS but not the specific implementation of `cp` that's being used. The problem can happen when the OS is Darwin **and** the `cp` command in the `PATH` does not support the `-X` flag (e.g. GNU's `cp`).

The use of temporary files test was chosen since macOS's native `cp` does not provide a `--help` or `--version` command to use instead. It does print its usage on wrong input, but it also emits a non-zero exit code, which is a problem because `script_phases.sh` has `pipefail` set. 

In short, making another test is a bit convoluted and I'm not sure if I have the bandwidth to come up with a working solution for that.

Comment (huntie) [reply packages/react-native/scripts/react_native_pods_utils/script_phases.sh:108]: I think perhaps we can replace this with the EdenFS detection we use in [Codegen's `build.sh`](https://github.com/facebook/react-native/blob/a4d6be89083598ea4fba8f07699fe905d0c2d163/packages/react-native-codegen/scripts/oss/build.sh#L33) — this way it's more clear why we are applying this flag against a bulk copy.

I'm happy to test this once imported.

```sh
    # Detect if we are in an EdenFS checkout
    if [ -x "$(command -v eden)" ] && eden info >/dev/null 2>&1; then
        CP_FLAGS="-X"
    else
        CP_FLAGS=""
    fi
```

**Comment by sebastiancarlos** [comment]
Looks like this was done already: https://github.com/blakef/react-native/commit/e70076a23409b9a4920e4c6b2d7d3f840e738cc5

**Comment by bsoft-Paulraj** [comment]
[!] CocoaPods could not find compatible versions for pod "React-NativeModulesApple":
  In Podfile:
    React-NativeModulesApple (from `../node_modules/react-native/ReactCommon/react/nativemodule/core/platform/ios`)

Specs satisfying the `React-NativeModulesApple (from `../node_modules/react-native/ReactCommon/react/nativemodule/core/platform/ios`)` dependency were found, but they required a higher minimum deployment target.

how to fix this issue
